spark-board-ui/src/App.jsx

```diff
@@ -66,20 +75,35 @@ export default function App() {
         }
     }, [nodesInitialized, settings]);
 
+    useEffect(() => {
+        setNodes(nodes.map(node => {
+            if (node.type == "transformation") {
+                node.position = transformationNodeCurrentPositions.get(node.id);
+            }
+            return node;
+        }));
+    }, [transformationNodeCurrentPositions]);
```

Comment:
I don't understand why this would fix the problem. What's the difference between calling `setNodes` here instead of the `useDagAnimation` hook?

Reply:
The problem is that the old `setNodes` was using the nodes state with the value that was received in the `useDagAnimation` parameter, so we had a race-condition when the onFrame was executed in parallel to the effect that updates the node sizes.
Making the `onFrame` to set the new positions' state and the new effect to update the nodes makes that the two effects aren't executed in parallel.
Another solution would be to pass a lambda expression to the `setNodes` inside `onFrame`, but I liked more this solution